src/CMakeLists.txt

```diff
@@ -165,163 +52,210 @@ execute_process(COMMAND git rev-parse HEAD
-
-message(STATUS "PYTHON_EXECUTABLE   = ${PYTHON_EXECUTABLE}")
-message(STATUS "PYTHON_INCLUDE_DIRS = ${PYTHON_INCLUDE_DIRS}")
-message(STATUS "PYTHON_LIBRARIES    = ${PYTHON_LIBRARIES}")
-
-
-#
-# Python support
-#
-set(src_py_support_files
-    nupic/py_support/NumpyArrayObject.cpp
-    nupic/py_support/NumpyVector.cpp
-    nupic/py_support/PyArray.cpp
-    nupic/py_support/PyHelpers.cpp
-    nupic/py_support/PythonStream.cpp
-    nupic/bindings/PySparseTensor.cpp
-)
-
-
-if(${NEEDS_BOOST})
-  message(STATUS "Hints for find_package()")
-  message(STATUS "  BOOST_ROOT  = ${BOOST_ROOT}")
-  if(MSVC OR MINGW)
-    # For windows, library name has lots of fields that must be matched.
-    # info about "--layout versioned" https://stackoverflow.com/questions/32991736/boost-lib-naming-are-missing/52208574#52208574
-    # filenames created by boost: https://www.boost.org/doc/libs/1_68_0/more/getting_started/unix-variants.html#library-naming
-    set(Boost_USE_STATIC_RUNTIME     OFF) 
-    set(Boost_USE_STATIC_LIBS        ON)  # only find static libs
-    set(Boost_USE_MULTITHREADED      ON)
-  endif()
-  set(Boost_DEBUG ON)
-  set(Boost_DETAILED_FAILURE_MSG ON)
   
-  find_package(Boost 1.56.0 REQUIRED COMPONENTS system filesystem)
-  if(${Boost_FOUND})
-	include_directories(${Boost_INCLUDE_DIRS})
-	message(STATUS "  Boost_INCLUDE_DIRS        = ${Boost_INCLUDE_DIRS}")
-	message(STATUS "  Boost_LIBRARIES           = ${Boost_LIBRARIES}")
-  else()
-  	message(FATAL_ERROR "Boost not found.")
-  endif()
-endif()
-message(STATUS "SWIG_EXECUTABLE           = ${SWIG_EXECUTABLE}")
-message(STATUS "SWIG_DIR                  = ${SWIG_DIR}")
-
-#
-# Common system libraries for shared libraries and executables
-#
-set(src_common_os_libs ${extra_lib_for_filesystem})
-
-if("${PLATFORM}" STREQUAL "linux")
-  list(APPEND src_common_os_libs pthread dl)
-elseif("${PLATFORM}" STREQUAL "darwin")
-  list(APPEND src_common_os_libs c++abi)
-elseif(MSYS OR MINGW)
-  list(APPEND src_common_os_libs psapi ws2_32 wsock32 rpcrt4)
-elseif("${PLATFORM}" STREQUAL "windows")
-  list(APPEND src_common_os_libs oldnames.lib psapi.lib ws2_32.lib)
-endif()
-
-message(STATUS "src_common_os_libs        = ${src_common_os_libs}")
+include_directories(SYSTEM
+                    ${PROJECT_BINARY_DIR}
+					${PROJECT_SOURCE_DIR}
+					${REPOSITORY_DIR}/external/common/include
+					${yaml-cpp_INCLUDE_DIRS}
+					${Boost_INCLUDE_DIRS}
+					)
 
 
 #
-# Setup nupic_core_solo static library, consisting of our own sources;
+# Setup nupic_core static library, consisting of our own sources and yaml libraries;
 #
 # this is an intermediate archive that will be merged with external static
 # libraries in a subsequent step to produce the nupic_core "combined" static
 # library.
 #
-set(src_lib_static_nupiccore_solo nupic_core_solo)
+set(src_lib_static_nupiccore_solo nupiccore_solo)
+
 
-set(src_nupiccore_srcs
+set(algorithm_files
     nupic/algorithms/Anomaly.cpp
+    nupic/algorithms/Anomaly.hpp
```

Comment:
do you need to include all? I think only cpp were listed and headers were picked by "grep *hpp" elsewhere.